lib/matplotlib/tests/test_figure.py

```diff
@@ -83,7 +84,10 @@ def test_gca():
 
     # the final request for a polar axes will end up creating one
     # with a spec of 111.
-    assert_true(fig.gca(polar=True) is not ax3)
+    with warnings.catch_warnings(record=True) as w:
```

Comment:
Within this context, you also need a `warnings.simplefilter('always')` or else the warning may not be triggered.